src/types/Typechecker/Suggestable.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Typechecker.Suggestable (
+                                Suggestable
+                                ,smallSuggest
+                                ,longSuggest
+                                )where
+
+-- Library dependencies
+import Text.PrettyPrint.Annotated
+import Text.Printf (printf)
+import Data.Maybe
+
+-- Module dependencies
+import AST.AST
+import AST.PrettyPrinter hiding (indent)
+import Typechecker.TypeError
+import Typechecker.Environment
+import Typechecker.Util
+import Identifiers
+import Types
+
+
+pipe = char '|'
+
+
+makeNotation :: Doc TCStyle
+makeNotation = styleLogistic (pipe $+$ equals) <+> styleDesc (text "note:")
+
+-- How to determine if to use a smallSuggest or longSuggest:
+-- If a problem justifies it, you could use both, 
+-- they are made so that from none to both are able to be used at the same time.
+--
+-- a smallSuggest are inlined with the highlighting of an error,
+-- therefore it is good practice for the text to be fairly short,
+-- about 32 characters seem to be a good maximum to strive for.
+-- If more are needed, use longSuggest instead.
+class Suggestable a where
+    smallSuggest :: a -> Doc ann
+    longSuggest :: a -> Doc TCStyle
+
+instance Suggestable TCError where
+    smallSuggest (TCError (NonAssignableLHSError) _) = "Can only be used on var or fields"
+    smallSuggest (TCError (MethodNotFoundError name ty) env)
+        | isMethodNameAFunction name ty env = text $ printf "Did you mean function `%s`?" (show name)
+    smallSuggest _ = empty
+
+    longSuggest (TCError (TypeMismatchError actual expected) _) =
```

Comment:
Maybe the first two cases of longSuggest can be refactored using guards. They are an identical copy except from the error type/

Reply:
As a TypeWithCapabilityMismatchError has a very "ugly" description I think it might be more wise to remove it entirely from suggestible as a way of subtly informing that this error needs rework.